website/source/docs/providers/aws/r/elastic_beanstalk_environment.html.markdown

```diff
@@ -0,0 +1,82 @@
+---
+layout: "aws"
+page_title: "AWS: aws_elastic_beanstalk_environment"
+sidebar_current: "docs-aws-resource-elastic-beanstalk-environment"
+description: |-
+  Provides an Elastic Beanstalk Environment Resource
+---
+
+# aws\_elastic\_beanstalk\_<wbr>Environment
+
+Provides an Elastic Beanstalk Environment Resource. Elastic Beanstalk allows 
+you to deploy and manage applications in the AWS cloud without worrying about 
+the infrastructure that runs those applications.
+
+Environments are often things such as `development`, `integration`, or 
+`production`.
+
+## Example Usage
+
+
+```
+resource "aws_elastic_beanstalk_application" "tftest" {
+  name = "tf-test-name"
+  description = "tf-test-desc"
+}
+
+resource "aws_elastic_beanstalk_environment" "tfenvtest" {
+  name = "tf-test-name"
+  application = "${aws_elastic_beanstalk_application.tftest.name}"
+  solution_stack_name = "64bit Amazon Linux 2015.03 v2.0.3 running Go 1.4"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) A unique name for the this Environment. This name is used 
+  in the application URL
+* `application` – (Required) Name of the application that contains the version 
+  to be deployed
+* `description` - (Optional) Short description of the Environment 
+* `cname` – (Optional) If specified, Elastic Beanstalk will try to use this
+  as a prefix. If omitted, an auto-generated prefix is automatically added
```

Comment:
Ah, gotcha. That should be clarified in the documentation. It would also be a great feature to have as part of this PR, if that isn't too hard to add. Thanks again.